app/main.py

```diff
@@ -1,3 +1,47 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: int | float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Distance | float) -> Distance:
+        if isinstance(other, Distance):
+            self.km += other.km
+        else:
+            self.km += other
+        return self
+
+    def __iadd__(self, other: Distance | float) -> Distance:
+        if isinstance(other, Distance):
+            self.km += other.km
+        else:
+            self.km += other
+        return self
+
+    def __mul__(self, other: Distance | float) -> Distance:
+        return Distance(self.km * other)
+
+    def __truediv__(self, other: Distance | int) -> Distance:
+        return Distance(round(self.km / other, 2))
+
+    def __lt__(self, other: Distance | float) -> bool:
+        return self.km < other
+
+    def __gt__(self, other: Distance | float) -> bool:
+        return self.km > other
+
+    def __eq__(self, other: Distance | float) -> bool:
+        return self.km == other
+
+    def __le__(self, other: Distance | float) -> bool:
+        return self.km <= other
+
+    def __ge__(self, other: Distance | float) -> bool:
+        return self.km >= other
```

Comment:
what if other will be a Distnace instance ? 
![image](https://user-images.githubusercontent.com/103201010/217558778-6c248b25-5ef6-471b-86d0-e9846e6f9cea.png)